API Gateway v2 acceptance tests: Replace `nodejs10.x` in lambda function configurations

<!---
Please note the following potential times when an issue might be in Terraform core:

* [Configuration Language](https://www.terraform.io/docs/configuration/index.html) or resource ordering issues
* [State](https://www.terraform.io/docs/state/index.html) and [State Backend](https://www.terraform.io/docs/backends/index.html) issues
* [Provisioner](https://www.terraform.io/docs/provisioners/index.html) issues
* [Registry](https://registry.terraform.io/) issues
* Spans resources across multiple providers

If you are running into one of these scenarios, we recommend opening an issue in the [Terraform core repository](https://github.com/hashicorp/terraform/) instead.
--->

<!--- Please keep this note for the community --->

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

<!--- Thank you for keeping this note for the community --->

```
=== RUN   TestAccAPIGatewayV2Integration_lambdaHTTP
=== PAUSE TestAccAPIGatewayV2Integration_lambdaHTTP
=== CONT  TestAccAPIGatewayV2Integration_lambdaHTTP
integration_test.go:353: Step 1/2 error: Error running apply: exit status 1
Error: error creating Lambda Function (1): InvalidParameterValueException: The runtime parameter of nodejs10.x is no longer supported for creating or updating AWS Lambda functions. We recommend you use the new runtime (nodejs14.x) while creating or updating functions.
{
RespMetadata: {
StatusCode: 400,
RequestID: "c9b3160c-6498-4c53-a112-ae607bba4400"
},
Message_: "The runtime parameter of nodejs10.x is no longer supported for creating or updating AWS Lambda functions. We recommend you use the new runtime (nodejs14.x) while creating or updating functions.",
Type: "User"
}
with aws_lambda_function.test,
on terraform_plugin_test.tf line 41, in resource "aws_lambda_function" "test":
41: resource "aws_lambda_function" "test" {
--- FAIL: TestAccAPIGatewayV2Integration_lambdaHTTP (19.81s)
```

```console
% find internal -name '*.go' -print | xargs grep 'nodejs10.x'
internal/service/apigatewayv2/authorizer_test.go:  runtime       = "nodejs10.x"
internal/service/apigatewayv2/integration_test.go:  runtime       = "nodejs10.x"
internal/service/lambda/function_test.go:  runtime       = "nodejs10.x"
internal/service/lambda/layer_version_test.go:  compatible_runtimes = ["nodejs12.x", "nodejs10.x"]
```

--- a/internal/service/apigatewayv2/authorizer_test.go
+++ b/internal/service/apigatewayv2/authorizer_test.go
@@ -429,7 +429,7 @@ resource "aws_lambda_function" "test" {
   function_name = %[1]q
   role          = aws_iam_role.iam_for_lambda.arn
   handler       = "index.handler"
-  runtime       = "nodejs10.x"
+  runtime       = "nodejs14.x"
 }
 
 resource "aws_iam_role" "test" {

--- a/internal/service/lambda/function_test.go
+++ b/internal/service/lambda/function_test.go
@@ -578,14 +578,14 @@ func TestAccLambdaFunction_versionedUpdate(t *testing.T) {
 				PreConfig: func() {
 					timeBeforeUpdate = time.Now()
 				},
-				Config: testAccVersionedNodeJs10xRuntimeConfig(path, funcName, policyName, roleName, sgName),
+				Config: testAccVersionedNodeJs14xRuntimeConfig(path, funcName, policyName, roleName, sgName),
 				Check: resource.ComposeTestCheckFunc(
 					testAccCheckFunctionExists(resourceName, funcName, &conf),
 					testAccCheckFunctionName(&conf, funcName),
 					acctest.CheckResourceAttrRegionalARN(resourceName, "arn", "lambda", fmt.Sprintf("function:%s", funcName)),
 					resource.TestCheckResourceAttr(resourceName, "version", versionUpdated),
 					acctest.CheckResourceAttrRegionalARN(resourceName, "qualified_arn", "lambda", fmt.Sprintf("function:%s:%s", funcName, versionUpdated)),
-					resource.TestCheckResourceAttr(resourceName, "runtime", lambda.RuntimeNodejs10X),
+					resource.TestCheckResourceAttr(resourceName, "runtime", lambda.RuntimeNodejs14X),
 					func(s *terraform.State) error {
 						return testAccCheckAttributeIsDateAfter(s, resourceName, "last_modified", timeBeforeUpdate)
 					},
@@ -2764,7 +2764,7 @@ resource "aws_lambda_function" "test" {
 `, layerName, funcName)
 }
 
-func testAccVersionedNodeJs10xRuntimeConfig(fileName, funcName, policyName, roleName, sgName string) string {
+func testAccVersionedNodeJs14xRuntimeConfig(fileName, funcName, policyName, roleName, sgName string) string {
 	return fmt.Sprintf(acctest.ConfigLambdaBase(policyName, roleName, sgName)+`
 resource "aws_lambda_function" "test" {
   filename      = "%s"
@@ -2772,7 +2772,7 @@ resource "aws_lambda_function" "test" {
   publish       = true
   role          = aws_iam_role.iam_for_lambda.arn
   handler       = "exports.example"
-  runtime       = "nodejs10.x"
+  runtime       = "nodejs14.x"
 }
 `, fileName, funcName)
 }

--- a/internal/service/lambda/layer_version_test.go
+++ b/internal/service/lambda/layer_version_test.go
@@ -357,7 +357,7 @@ resource "aws_lambda_layer_version" "lambda_layer_test" {
   filename   = "test-fixtures/lambdatest.zip"
   layer_name = "%s"
 
-  compatible_runtimes = ["nodejs12.x", "nodejs10.x"]
+  compatible_runtimes = ["nodejs12.x", "nodejs14.x"]
 }
 `, layerName)
 }

